src/libinjector/injector.c

```diff
@@ -838,15 +839,39 @@ event_response_t mem_callback(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
     return VMI_EVENT_RESPONSE_SET_REGISTERS;
 }
 
-event_response_t cr3_callback(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+event_response_t cr3_catch_create_proc(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 {
+    struct injector* injector = info->trap->data;
+    (void)drakvuf;
 
+    PRINT_DEBUG("CR3 in catcher changed to 0x%" PRIx64 ". PID: %u PPID: %u\n",
+                info->regs->cr3, info->proc_data.pid, info->proc_data.ppid);
+
+    if ( info->proc_data.ppid == injector->target_pid )
+    {
+        char *pname = drakvuf_get_process_name(drakvuf, info->proc_data.base_addr, 0);
```

Comment:
An easy work-around that is that if you need an application to be running with the same name as you will inject is to start it via another process - so that it's ppid is different.